web/src/pages/project/ProjectSummary.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react'
+import { useParams, useNavigate, useSearchParams } from 'react-router-dom'
+
+import { Dropdown } from 'semantic-ui-react'
+import ProjectSelector from './ProjectSelector'
+import { WebApi, ProjectSummaryResponse } from '../../sm-api/api'
+
+import PageOptions from './PageOptions'
+import SeqrLinks from './SeqrLinks'
+import MultiQCReports from './MultiQCReports'
+import SummaryStatistics from './SummaryStatistics'
+import BatchStatistics from './BatchStatistics'
+import ProjectGrid from './ProjectGrid'
+
+const PAGE_SIZES = [20, 40, 100, 1000]
+
+const ProjectSummary: React.FunctionComponent = () => {
+    const navigate = useNavigate()
+
+    const { projectName, page } = useParams()
+
+    const [searchParams] = useSearchParams()
+    const pageSize = searchParams.get('size') || 20
+    let directToken = 0
+    if (page && pageSize) {
+        directToken = +page * +pageSize - +pageSize
+    }
+
+    const validPages = !!(page && +page && pageSize && +pageSize && PAGE_SIZES.includes(+pageSize))
+
+    const [summary, setSummary] = React.useState<ProjectSummaryResponse | undefined>()
+    const [pageNumber, setPageNumber] = React.useState<number>(validPages ? +page : 1)
+    const [isLoading, setIsLoading] = React.useState<boolean>(false)
+    const [error, setError] = React.useState<string | undefined>()
+    const [pageLimit, _setPageLimit] = React.useState<number>(
+        validPages ? +pageSize : PAGE_SIZES[0]
+    )
+
+    const handleOnClick = React.useCallback(
+        (p) => {
+            navigate(`/project/${projectName}/${p}?size=${pageLimit}`)
+            setPageNumber(p)
+        },
+        [navigate, pageLimit, projectName]
+    )
+
+    const getProjectSummary = React.useCallback(
+        async (token: any) => {
+            if (!projectName) {
+                setSummary(undefined)
+                return
+            }
+            const sanitisedToken = token || undefined
+            setError(undefined)
+            setIsLoading(true)
+            new WebApi()
```

Comment:
Just a comment - I find it much cleaner and idiomatic to use the `await` syntax inside `async` functions instead of `then/catch` promises. So this promise becomes:

```ts
try {
    const response = await new WebApi().getProjectSummary(projectName, pageLimit, sanitisedToken)
    setSummary(resp.data)
} catch (er) {
    setError(er.message)
} finally {
    setIsLoading(false)
}
```

Reply:
Fixed, although this will migrate to GraphQL soon